tests/forms_tests/widget_tests/test_select.py

```diff
@@ -313,32 +342,43 @@ def test_doesnt_localize_option_value(self):
         ]
         html = """
         <select name="time">
-        <option value="00:00:00">midnight</option>
-        <option value="12:00:00">noon</option>
+          <option value="00:00:00">midnight</option>
+          <option value="12:00:00">noon</option>
         </select>
         """
         self.check_html(self.widget(choices=choices), "time", None, html=html)
 
-    def test_options(self):
-        options = list(
-            self.widget(choices=self.beatles).options(
-                "name",
-                ["J"],
-                attrs={"class": "super"},
-            )
+    def test_render_as_subwidget(self):
+        """A RadioSelect as a subwidget of MultiWidget."""
+        choice_beatles = (("", "------"),) + self.beatles
+        choice_pantera = (("", "------"),) + self.pantera
+        choices = [choice_beatles, choice_pantera]
+
+        self.check_html(
+            MultiWidget(
+                [self.widget(choices=choices[0]), self.widget(choices=choices[1])]
+            ),
+            "bands",
+            ["J", "Da"],
+            html=(
+                """
+                <select name="bands_0">
+                  <option value="">------</option>
+                  <option value="J" selected>John</option>
+                  <option value="P">Paul</option>
+                  <option value="G">George</option>
+                  <option value="R">Ringo</option>
+                </select>
+                <select name="bands_1">
+                  <option value="">------</option>
+                  <option value="Ph">Phil</option>
+                  <option value="Re">Rex</option>
+                  <option value="Da" selected>Darrel</option>
+                  <option value="Vi">Vinni</option>
+                </select>
+            """
+            ),
```

Comment:
I also added `test_render_as_subwidget` for `test_select`. I expanded those tests for `test_select` and `test_radioselect` a little bit since those testcases are checking correctness of `MultiWidget`, let me know what you think :)

Reply:
I'm not sure why we need another band for testing the `MultiWidget` case, could you share your rationale please?

Reply:
I assumed, since this widget is supporting generation of multiple widgets, it would be nice to show it is accepting more arguments. Maybe I should split existing band into smaller pieces instead of adding new one. If it seems unnecessary, let me know, I'll change it right away :)

Reply:
Ah, I see! I understand your goal. But, I was thinking that we could indeed have another widget in there to fully exercise the "multiple" part of the `MultiWidget`, but perhaps we could use a simpler widget like a `TextInput`.

Do you see a concrete gain from having two instances of the same widget in a `MultiWidget`? Or do you think that having at least 2 widgets, where at least one is the one we care is enough?

Reply:
I'm gonna replace the second widget with a simpler one and remove the newly added band. Then, likely I'll merge this! :woman_dancing:

Reply:
I pushed new changes 5 minutes ago

Reply:
Oh, sorry, I just overwrote those. I'm looking at the diff now, and I think your changes were not up to date with the modifications I made earlier this week. I applied a set of styling improvements and minor cleanups, so if you want to do another change, please update your branch with pulling my changes in.

Wanna review the latest that I pushed, and let me know what you think? I'll wait until end-of-day to merge this.

Reply:
No problem, context is the same as my pushed changes :smile:

Reply:
> Only difference is that I changed `bands` to `band` after deleting second band. But I think such small difference is not worth changing smiley

Great point, thanks for mentioning it! I'll restore the original name for the sake of diff tidiness. Will merge soon!